app/src/main/java/org/jellyfin/mobile/player/PlayerViewModel.kt

```diff
@@ -408,6 +417,46 @@ class PlayerViewModel(application: Application) : AndroidViewModel(application),
         }
     }
 
+    private fun applyMediaSegments(jellyfinMediaSource: JellyfinMediaSource) {
+        viewModelScope.launch {
+            if (jellyfinMediaSource.item != null) {
+                val mediaSegments = runCatching {
+                    mediaSegmentRepository.getSegmentsForItem(jellyfinMediaSource.item)
+                }.getOrNull().orEmpty()
+
+                for (mediaSegment in mediaSegments) {
+                    val action = mediaSegmentRepository.getMediaSegmentAction(mediaSegment)
+
+                    when (action) {
+                        MediaSegmentAction.SKIP -> addSkipAction(mediaSegment)
+                        MediaSegmentAction.NOTHING -> Unit
+                        // Unimplemented
+                        MediaSegmentAction.ASK_TO_SKIP -> Unit
+                    }
+                }
+            }
+        }
+    }
+
+    private fun addSkipAction(mediaSegment: MediaSegmentDto) {
+        val player = playerOrNull ?: return
+
+        player.createMessage { _, _ ->
+            // We can't seek directly on the ExoPlayer instance as not all media is seekable
+            // the seek function in the PlaybackController checks this and optionally starts a transcode
+            // at the requested position
+            // TODO: The above is probably true for jellyfin-android as well.
+            // But I believe there is no such logic here.
```

Comment:
Correct, the seekTo you invoke now is directly on ExoPlayer. I think that's fine for now though as the mobile app doesn't support "transcode on unsupported seek" functionality yet.